tests/0102-static_group_rebalance.c

```diff
@@ -0,0 +1,217 @@
+        test_conf_set(conf, "auto.offset.reset", "earliest");
+        test_conf_set(conf, "topic.metadata.refresh.interval.ms", "500");
+
+        rd_kafka_conf_set_opaque(conf, &c[0]);
+        test_conf_set(conf, "group.instance.id", "consumer1");
+        c[0].rk = test_create_consumer("static", &rebalance_cb,
+                                       rd_kafka_conf_dup(conf), NULL);
+
+        rd_kafka_conf_set_opaque(conf, &c[1]);
+        test_conf_set(conf, "group.instance.id", "consumer2");
+        c[1].rk = test_create_consumer("static", &rebalance_cb,
+                                       rd_kafka_conf_dup(conf), NULL);
+
+        c[0].max_rebalance_cnt = 1;
+        c[1].max_rebalance_cnt = 1;
+
+        test_consumer_subscribe(c[0].rk, rd_strdup(tsprintf("^%s.*", topic)));
+        test_consumer_subscribe(c[1].rk, rd_strdup(tsprintf("^%s.*", topic)));
+
+        test_create_topic(NULL, topic, 3, 1);
+        test_produce_msgs_easy(topic, test_id_generate(), 0, msgcnt);
+
+        TEST_SAY("Waiting for partition assignment\n");
+        await_assignment(&c[0]);
+        await_assignment(&c[1]);
+
+        TEST_SAY("Bouncing c[1] instance.\n");
+        test_consumer_close(c[1].rk);
+        rd_kafka_destroy(c[1].rk);
+
+        /** 
+         * Removeing and adding a new member with the same group instance id 
+         * should not prompt a rebalance for the original remaining member.
+         */
+        c[1].max_rebalance_cnt++;
+        c[1].rk = test_create_consumer("static", &rebalance_cb,
+                                       conf, NULL);
+        test_consumer_subscribe(c[1].rk, rd_strdup(tsprintf("^%s.*", topic)));
+
+        /* Wait for c[1] to rejoin, serve c[0] ops */
+        await_assignment(&c[1]);
+        do_consume(&c[0], 5/*5s*/);
+
+        TEST_SAY("Creating a new topic.\n");
+        /* Expanding the subscription forces a rebalance */
+        c[0].max_rebalance_cnt++;
+        c[1].max_rebalance_cnt++;
+
+        /* The topic prefix uses the test id which is "random" */
+        test_create_topic(c[0].rk, rd_strdup(tsprintf("%snew", topic)), 1, 1);
+        do_consume(&c[0], 5/*5s*/);
+        do_consume(&c[1], 5/*5s*/);
+        await_assignment(&c[0]);
+
+        /* Wait until session.timeout.ms is exceeded to force a rebalance. */
+        TEST_SAY("Closing c[1], waiting for static instance to be evicted.\n");
+        c[0].max_rebalance_cnt++;
+
+        test_consumer_close(c[1].rk);
+        rd_kafka_destroy(c[1].rk);
+
+        /* 2x heartbeat interval to give time for c[0] to recognize rebalance */
+        rd_sleep(6);
+        do_consume(&c[0], 2/*1s*/);
+
+        TEST_ASSERT(c[0].assign_cnt == c[0].max_rebalance_cnt,
+                    "c[0] rebalanced %d times, expected %d",
+                    c[0].assign_cnt, c[0].max_rebalance_cnt);
+
+        TEST_SAY("Closing remaining consumers\n");
+        test_consumer_close(c[0].rk);
+        rd_kafka_destroy(c[0].rk);
+
```

Comment:
conf is leaked

Reply:
conf is handed over (without a call to dup) to the last instance of c1